tools/licensegen/licensegen.go

```diff
@@ -6,24 +6,26 @@ import (
 	"log"
 	"os"
 	"path/filepath"
+	"strings"
 )
 
 func main() {
-	err := filepath.Walk(
-		"./vendor",
-		func(path string, info os.FileInfo, err error) error {
-			if filepath.Base(path) == "LICENSE" {
+	err := filepath.Walk("./vendor", func(path string, _ os.FileInfo, _ error) error {
+		base := filepath.Base(path)
+		ext := filepath.Ext(base)
+		if strings.TrimSuffix(base, ext) == "LICENSE" {
```

Comment:
`strings.HasPrefix("LICENSE")` could also work so we don't have to update the list of extensions in the future